Rmd/go_pathway_report.Rmd

```diff
@@ -443,23 +444,26 @@ for (i in seq_along(contrasts)) {
 This section uses clusterProfiler `gseKEGG()` to detect enriched KEGG pathways and enrichplot `gseaplot2` to plot running scores.
 
 ```{r 'gsea_analysis_by_contrast_make_plots', collapse=TRUE, eval=params$run_pathway_analysis, include=params$run_pathway_analysis,  warning = FALSE, message = FALSE}
+
 plotListGSEA <- list()
 plotListGSEA_escore <- list()
 contrasts <- levels(allResults$contrast)
+bg_genes <-read.gmt("~/storage/LearnPipline/R-ODAF_Health_Canada/c2.cp.kegg_medicus.v2023.2.Hs.symbols.gmt") 
+
 for (i in seq_along(contrasts)) {
   DEGs_full <- dplyr::left_join(allResults %>% filter(contrast ==  contrasts[i]),
                                 id_table_entrez,
                                 by = "Feature_ID")
   foldChanges <- DEGs_full %>% dplyr::pull(log2FoldChange)
-  names(foldChanges) <- DEGs_full %>% dplyr::pull(Entrez_ID)
+  names(foldChanges) <- DEGs_full %>% dplyr::pull(Gene_Symbol.x)
```

Comment:
Have been speaking to Lauren about making a generic path for the gmt for KEGGpathways, however, I come across the issue "Error in knit_params_get(input_lines, params): render params not declared in YAML: KEGGpathways_filename". I have included "KEGGpathways_filename" both in the config params as well as the rmd params. While trying to wrap my head around why this is, I have found a few things: if you include a header "render" in the config params and put KEGGpathways_filename there it will run. If you only have the params in the rmd but not the config, it will run. To address the gene_symbol.x, this was one of the features I worked on at the very beginning of starting so my memory is a bit hazy, but I was unable to link using just gene_symbol as it was not present. Now that I have a greater understanding of the linkages I can take a look at this again

Reply:
You are right! both allResults and id_table_entrez have gene symbol. I checked using anti_join() and found there is no difference between the two. I have decided to remove one of them and rename the remaining to "Gene_Symbol" with the .x to remove the redundancy.

Reply:
If you add gene symbol as an argument to the join it should actually use the information in those columns to make sure the two tables are linked by the same genes, does that make sense?

Reply:
Thank you that is much better! I have included it